rerun_py/rerun_sdk/rerun/time.py

```diff
@@ -101,11 +101,18 @@ def set_time(
         )
     elif timestamp is not None:
         nanos = to_nanos_since_epoch(timestamp)
-        bindings.set_time_timestamp_nanos_since_epoch(
-            timeline,
-            nanos,
-            recording=recording.to_native() if recording is not None else None,
-        )
+        try:
+            bindings.set_time_timestamp_nanos_since_epoch(
+                timeline,
+                nanos,
+                recording=recording.to_native() if recording is not None else None,
+            )
+        except OverflowError as err:
+            raise ValueError(
+                f"set_time: `timestamp={timestamp!r}` is out of range; "
+                f"expected seconds since Unix epoch, datetime.datetime, or numpy.datetime64 "
+                f"(timeline='{timeline}')"
+            ) from err
```

Comment:
How large integers were you passing in, and why?

And shouldn't we protect against the same problem for sequence and duration timelines too?

Reply:
I was passing a timestamp in milliseconds since unix epoch, e.g. '1_745_943_196_000'. But since we assume seconds, it would then be multiplied by 1e9 to convert to nanoseconds.

I don't think sequence and duration suffer from the same gotcha, so I'd argue it's not necessary.

Reply:
Yea I like this more as well. However `2_000_000_000_000` is only valid for about 8 years. 

Let's check if timestamp is bigger than `1e11` which will always mean milliseconds instead of seconds 👍